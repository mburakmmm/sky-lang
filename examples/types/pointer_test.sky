# Pointer type test

function printPointer(ptr: any)
  print("Pointer type annotation works!")
  print("Value passed:")
  print(ptr)
end

function takesPointer(x: any): any
  return x
end

function main
  print("=== Pointer Type Syntax Test ===")
  
  let x = 42
  printPointer(x)
  
  print("Pointer type in function signature:")
  let ptrResult = takesPointer(x)
  print(ptrResult)
  
  print("=== Note ===")
  print("Pointer syntax supported in type annotations")
  print("Full pointer ops require unsafe blocks + LLVM")
  
  print("=== Done ===")
end

