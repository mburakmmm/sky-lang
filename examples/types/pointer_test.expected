=== Pointer Type Syntax Test ===
Pointer type annotation works!
Value passed:
42
Pointer type in function signature:
42
=== Note ===
Pointer syntax supported in type annotations
Full pointer ops require unsafe blocks + LLVM
=== Done ===

